cmd/exec_test.go

```diff
@@ -147,3 +148,236 @@ func TestExecBundleFlag(t *testing.T) {
 
 	})
 }
+
+func TestInvalidConfig(t *testing.T) {
+	var buf bytes.Buffer
+	params := exec.NewParams(&buf)
+	params.Fail = true
+	params.FailDefined = true
+
+	err := exec.Exec(context.TODO(), nil, params)
+	if err == nil || err.Error() != "specify --fail or --fail-defined but not both" {
+		t.Fatalf("Expected error '%s' but got '%s'", "specify --fail or --fail-defined but not both", err.Error())
+	}
+}
+
+func TestFailFlagCases(t *testing.T) {
+
+	var tests = []struct {
+		description string
+		files       map[string]string
+		decision    string
+		expectError bool
+		expected    interface{}
+		fail        bool
+		failDefined bool
+	}{
+		{
+			description: "--fail-defined with undefined result",
+			files: map[string]string{
+				"files/test.json": `{"foo": 7}`,
+				"bundle/x.rego": `package system
+
+		test_fun := x {
+			x = false
+			x
+		}
+
+		undefined_test {
+			test_fun
+		}`,
+			},
+			decision:    "",
+			expectError: false,
```

Comment:
[nit] Let's omit the default fields. If we don't specify these two, they'll have the same values. Same goes for the other cases below.